lib/jsonapi/utils/response/formatters.rb

```diff
@@ -194,6 +194,10 @@ def result_options(records, options)
             if JSONAPI.configuration.top_level_meta_include_record_count
               data[:record_count] = count_records(records, options)
             end
+
+            if JSONAPI.configuration.top_level_meta_include_page_count
+              data[:page_count] = count_pages(records, options)
```

Comment:
What about to pass `data[:record_count]` to `#count_pages` so that it can skip querying a `COUNT()` to the  database again?

```ruby
data[:page_count] = count_pages(data[:record_count])
```